contributors/design-proposals/controller_history.md

```diff
@@ -0,0 +1,444 @@
+# Controller History
+
+**Author**: kow3ns@
+
+**Status**: Proposal
+
+## Abstract 
+In Kubernetes, in order to update and rollback the configuration and binary 
+images of controller managed Pods, users mutate DaemonSet, StatefulSet, 
+and Deployment Objects, and the corresponding controllers attempt to transition 
+the current state of the system to the new declared target state. 
+
+To facilitate update and rollback for these controllers, and to provide a 
+primitive that third party controllers can build on, we propose a mechanism 
+that allows controllers to manage a bounded history of revisions to the declared 
+target state of their generated Objects.
+
+## Affected Components
+
+1. API Machinery
+1. API Server
+1. Kubectl
+1. Controllers that utilize the feature
+
+## Requirements
+
+1. History is a collection of points in time, and each point in time must be 
+represented by its own Object. While it is tempting to aggregate all of an 
```

Comment:
Event if the points in time are the same we will have separate Objects for them? I'm talking about situation, when changing image: nging -> httpd -> nginx